packages/core/src/new-api/internal/reconciliation/futures/reconcileNamedStaticCall.ts

```diff
@@ -0,0 +1,46 @@
+import { isEqual } from "lodash";
+
+import { StaticCallExecutionState } from "../../../types/execution-state";
+import { NamedStaticCallFuture } from "../../../types/module";
+import { ExecutionStateResolver } from "../execution-state-resolver";
+import { ReconciliationContext, ReconciliationFutureResult } from "../types";
+import { fail, resolveFromAddress, safeToString } from "../utils";
+
+export function reconcileNamedStaticCall(
+  future: NamedStaticCallFuture<string, string>,
+  executionState: StaticCallExecutionState,
+  context: ReconciliationContext
+): ReconciliationFutureResult {
+  const contractAddress: string =
+    ExecutionStateResolver.resolveContractToAddress(future.contract, context);
+
+  if (!isEqual(contractAddress, executionState.contractAddress)) {
+    return fail(
+      future,
+      `Contract address has been changed from ${executionState.contractAddress} to ${contractAddress}`
+    );
+  }
+
+  if (!isEqual(future.functionName, executionState.functionName)) {
+    return fail(
+      future,
+      `Function name has been changed from ${executionState.functionName} to ${future.functionName}`
+    );
+  }
+
+  if (!isEqual(future.args, executionState.args)) {
```

Comment:
Won't these comparison of `args` fail if they have a future or runtime value? Same for the ones that receive `constructorArgs`.

I think I'd try to avoid using `isEqual` unless its really necessary, as it can lead to this situations of incomplete comparison.

Reply:
I have pulled out the serialization replacement logic and used it to make the future comparison work with nested futures and runtime args.